TLM/TLM/Manager/Impl/LaneConnectionManager.cs

```diff
@@ -384,23 +392,18 @@ public class LaneConnectionManager
             return true;
         }
 
-        protected override void HandleInvalidSegment(ref ExtSegment seg) {
-            HandleInvalidSegmentImpl(seg.segmentId);
-        }
-
-        private void HandleInvalidSegmentImpl(ushort segmentId) {
+        private void ReleasingSegment(ushort segmentId, ref NetSegment segment) {
```

Comment:
maybe rename to `OnReleasingSegment()`?

Reply:
on the second thought this is fine too.
https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-type-members#names-of-events

CS uses the `On...` convention. 
Notifier uses events that begin with `On`.  also we have `OnSegmentChange/Modified/Replacement` and several other events like this.
In other places we don't use the `On...` naming convention. So its a mixed bag so far.

Reply:
In addition to the naming convention on the page you linked to, Microsoft's convention, at least as implemented in places like WinForms, is that "On..." is used on virtual methods that are called by event handlers. So basically, it's a second layer of event handling within the context of a specific class that monitors an event.